src/braket/default_simulator/linalg_utils.py

```diff
@@ -50,15 +45,22 @@ def multiply_matrix(
         return _multiply_matrix(state, matrix, targets)
 
     control_state = control_state or (1,) * len(controls)
-    num_qubits = len(state.shape)
-    control_slices = {i: _SLICES[state] for i, state in zip(controls, control_state)}
-    ctrl_index = tuple(
-        control_slices[i] if i in controls else _NO_CONTROL_SLICE for i in range(num_qubits)
-    )
-    state[ctrl_index] = _multiply_matrix(state[ctrl_index], matrix, targets)
+
+    ctrl_slices = [slice(None)] * len(state.shape)
+    for i, state_val in zip(controls, control_state):
+        ctrl_slices[i] = slice(None, 1) if state_val == 0 else slice(1, None)
+
+    state[tuple(ctrl_slices)] = _multiply_matrix(state[tuple(ctrl_slices)], matrix, targets)
```

Comment:
You _could_ still keep the three constants
```
_NEG_CONTROL_SLICE = slice(None, 1),
_CONTROL_SLICE = slice(1, None),
_NO_CONTROL_SLICE = slice(None, None),
```
rather than recreating them each time.

Reply:
Yeah I think I was trying to be too concise here.